### PR TITLE
Change ERR_add_error_[v]data to append

### PR DESCRIPTION
**Pull request by richsalz**
The "add error data" functions now append to the current error.
Add a test for this, and update the documentation.
Cleanup some of the ERR_put functions.
In the FIPS module, always append "(in the FIPS module)" to any errors.

This replaces #8887.  Ping @levitte.

### CODE REVIEW

File: crypto/err/err.c [hunk 1]
```diff
@@ -408,25 +408,6 @@ void ERR_put_error(int lib, int func, int reason, const char *file, int line)
...
-        /* Skip the "*POSIX(" prefix */
-        file += sizeof("*POSIX(") - 1;
-        end = &file[strlen(file) - 1];
-        if (*end == ')')
-            *end = '\0';
-        /* Optional: use the basename of the path only. */
-        if ((end = strrchr(file, '/')) != NULL)
-            file = &end[1];
-    }
-#endif
```
Comment (levitte) [inline crypto/err/err.c]: The reason you're removing this?  Please do note that we still have a config target for that system:

https://github.com/openssl/openssl/blob/7f02a0932c8c182bc2f6320748d81e86d650c1a6/Configurations/10-main.conf#L1240-L1251

So if you're suggesting that system should be cleaned away, I think that should be done in a separate PR.

Comment (richsalz) [reply crypto/err/err.c]: Okay separate PR in #9185

File: doc/man3/ERR_put_error.pod [hunk 2]
```diff
@@ -23,6 +23,7 @@ This function is usually called by a macro.
 ERR_add_error_data() associates the concatenation of its B<num> string
 arguments with the error code added last.
 ERR_add_error_vdata() is similar except the argument is a B<va_list>.
++Multiple calls to these functions append to the current top of the error queue.
```
Comment (bernd-edlinger) [inline doc/man3/ERR_put_error.pod]: there is a + character at the beginning of this line.

Comment (levitte) [reply doc/man3/ERR_put_error.pod]: Oops...

Comment (richsalz) [reply doc/man3/ERR_put_error.pod]: good eyes.  fixed.  if/when merged, i hope all commits will be squashed.

**Comment by bernd-edlinger** [comment]
> if/when merged, i hope all commits will be squashed.

you could use `git commit --fixup <hash> ` to assist
automatic squashing with `git rebase -i --autosquash` on merge.

**Comment by richsalz** [comment]
yes, but i was more concerned about the OSD stuff going and resurrecting.

**Comment by levitte** [comment]
Merged.

8908d18cb1020f225170dd9090206842dbb6bbe6 Change ERR_add_error_[v]data to append
